makeStage2.sh

```diff
@@ -1,6 +1,7 @@
 #!/bin/sh
 set -e
```

Comment:
Let's turn this into a `set -euo pipefail` (a.k.a [bash strict mode](http://redsymbol.net/articles/unofficial-bash-strict-mode/)) now that we're allowing bash in the build system. (obviously line 1 will need changing to `#!/bin/bash`

Reply:
If we're doing this, we can clean up the nasty `|| true`  clauses on lines 20--29 by using `nullglob`: `shopt -s nullglob`

Reply:
I have added the set -euo pipefail.

Reply:
The `|| true` is required on line 29 because there seems to be a problem with `poplibr` which never exits with status 0. So it is only the `|| true` that might be eligible for simplification via nullglob. But wouldn't the `mv` fail even with nullglob??

Reply:
Ah, that's disappointing (poplibr is a bad unix citizen!). 
Yes, you're quite right about the nullglob option! From the bash manual:

> If no matching filenames are found, and the shell option nullglob is  not
enabled,  the  word  is  left  unchanged.  If the nullglob option is set, and no
matches are found, the word is removed.

So if we had `nullglob` and no matches, the `mv` invocation would become `mv old`, which would error.

Reply:
I have adopted the set -euo pipefail everywhere it did not break the build. The catch is that some of the inherited scripts reference undefined variables (yikes!). So I am adopting this as [a tech-debt item](https://github.com/GetPoplog/Seed/issues/95).